src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -156,6 +160,12 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
             buildImageCmd = buildImageCmd.withBuildArg(arg, value)
         }
 
+        if (getCacheFrom()) {
+            // Workaround a bug in dockerjava that double-unmarshalls this argument
```

Comment:
Did you report this bug to Docker Java? I'd like to avoid adding workarounds in this plugin.

Reply:
Definitely open a big with those guys but I don't personally think this should stop things from moving forward. And given that they do releases once a blue moon I don't think we should hold things up on our end.

On a side note: we've been using spotifys docker-client here internally for writing a Jenkins shared library and it's really easy and great IMO. Its giving me more hope that switching things on our end wouldn't be so bad.

Reply:
I submitted a fix https://github.com/docker-java/docker-java/pull/1086